http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeServiceDbQueries.scala

```diff
@@ -306,411 +247,267 @@ object RuntimeServiceDbQueries {
-              #${whereFilterLabels}
-          ) AS LABEL_FILTERED
-          inner join `RUNTIME_CONFIG` RG on LABEL_FILTERED.runtimeConfigId = RG.`id`
-          left join `CLUSTER_PATCH` CP on LABEL_FILTERED.id = CP.`clusterId`
-          GROUP BY LABEL_FILTERED.id""".stripMargin
-
-    sqlStatement.as[ListRuntimeResponse2]
-  }
-
-  private def listRuntimesHelper(labelMap: LabelMap,
-                                 excludeStatuses: List[RuntimeStatus],
-                                 creatorOnly: Option[WorkbenchEmail],
-                                 workspaceId: Option[WorkspaceId],
-                                 cloudContextOrCloudProvider: Option[Either[CloudContext, CloudProvider]]
-  ): DBIO[Vector[ListRuntimeResponse2]] = {
-    val runtimeQueryFilteredByCreator = creatorOnly match {
-      case Some(creator) => List(s"C.`creator` = '${creator.value}'")
-      case None          => List.empty
-    }
-    val runtimeQueryFilteredByDeletion = excludeStatuses.map(s => s"C.`status` != '${s.toString}'")
-
-    val runtimeQueryFilteredByWorkspace = workspaceId match {
-      case Some(wid) =>
-        List(s"C.`workspaceId` = '${wid.value.toString}'")
-      case None => List.empty
-    }
-
-    val runtimeQueryFilteredByCloud = cloudContextOrCloudProvider match {
-      case Some(Left(CloudContext.Gcp(gp)))     => List(s"C.`cloudContext` = '${gp.value}'")
-      case Some(Left(CloudContext.Azure(actx))) => List(s"C.`cloudContext` = '${actx.asString}'")
-      case Some(Right(cloudProvider))           => List(s"C.`cloudProvider` = '${cloudProvider.asString}'")
-      case None                                 => List.empty
-    }
-
-    val clusterFilters =
-      (runtimeQueryFilteredByCreator ++ runtimeQueryFilteredByDeletion ++ runtimeQueryFilteredByWorkspace ++ runtimeQueryFilteredByCloud)
-        .mkString(" AND ")
+    val runtimesFiltered = labelMap
+      .foldLeft(runtimesFilteredSimple) {
+        case (someRuntimeQuery: Query[ClusterTable, ClusterRecord, Seq], (key: String, value: String)) =>
+          for {
+            (runtime, _) <- someRuntimeQuery join labelQuery on ((r, l) =>
+              l.resourceId === r.id &&
+                l.resourceType === LabelResourceType.runtime &&
+                l.key === key &&
+                l.value === value
+            )
+          } yield runtime: ClusterTable
+      }
 
-    val clusterFiltersFinal = if (clusterFilters.isEmpty) "" else s"where ${clusterFilters}"
+    // Assemble response
+    val runtimesJoined = runtimesFiltered
+      .join(runtimeConfigs)
+      .on((runtime, runtimeConfig) => runtime.runtimeConfigId === runtimeConfig.id)
+      .joinLeft(patchQuery)
+      .on((runtimeWithConfig, runtimePatch) => runtimeWithConfig._1.id === runtimePatch.clusterId)
+      .joinLeft(labelQuery)
+      .on((runtimeWithConfigWithPatch, l) =>
+        runtimeWithConfigWithPatch._1._1.id === l.resourceId &&
+          l.resourceType === LabelResourceType.runtime
+      )
```

Comment:
Adding more fields! The only one which "filters" is runtimeConfig, which is presumably because a runtime without a config is not sane.